### PR TITLE
Fetch things in parallel to make index load faster

### PR DESCRIPTION
**Pull request by ryankshaw**
Howdy Jason, 
Random internet stranger here. Ended up here from following a link you posted on twitter. Your site looks really slick! 

Saw this one example of how you could make your loader function faster. You probably already knew this and don't feel the need to actually merge this but just in case it was helpful I figured I'd make this PR

### CODE REVIEW

File: site/app/routes/index.jsx [hunk 1]
```diff
@@ -7,22 +7,21 @@ import { SectionHero } from '../components/section-hero.jsx';
...
+  const schedulePromise = fetchJson('https://www.learnwithjason.dev/api/schedule');
+  const featuredPromise = fetchJson('https://www.learnwithjason.dev/api/episodes/featured');
+  const sponsorsPromise = fetchJson('https://www.learnwithjason.dev/api/sponsors');
+
+  return { 
+    episodes: await episodesPromise,
+    schedule: await schedulePromise, 
+    featured: await featuredPromise, 
+    sponsors: await sponsorsPromise,
+  };
```
Comment (mehulkar) [inline site/app/routes/index.jsx:24]: Hi! I'm another internet stranger :). This code does not actually parallelize (unless Remix is doing some compiling under the hood that I don't know about!) If you want to parallelize, use

```js
await Promise.all([
  episodesPromise,
  schedulePromise,
  // ...etc
])
```

Here's a demo I made that hopefully helps: https://jsfiddle.net/69f1h28d/1/

Comment (ryankshaw) [reply site/app/routes/index.jsx:24]: actually, my code _does_ fetch things in parallel, I made another revision to your jsfiddle to show you
https://jsfiddle.net/rckx5Lms/2/

Comment (ryankshaw) [reply site/app/routes/index.jsx:24]: the tl;dr is that
```JavaScript

  return { 
     a: (await makePromise(500, 'a')),
     b: (await makePromise(100, 'b')),
   }

// is different than

  const cPromise =  makePromise(5000, 'c')
  const dPromise =  makePromise(1000, 'd')
  return { 
     c: await cPromise,
     d: await dPromise
   };
}
```
The difference is that that one `await`s on the same line as the promise is made, the other creates the 2 promises then `await`s them later. 

You're right that you can also use `Promise.all[` but that is more ceremony and less readable

anyway, feel free to take my suggestion. no worries if you don't. still a really cool site and I think remix is pretty neat!

Comment (mehulkar) [reply site/app/routes/index.jsx:24]: Huh, TIL! Thanks for the explanation 😃 

(PS, not my site I just popped in here on idle time (...and exercise [Cunningham's Law](https://meta.wikimedia.org/wiki/Cunningham%27s_Law) apparently!))

Comment (jlengstorf) [reply site/app/routes/index.jsx:24]: oh dang! I, too, thought you had to use `Promise.all` for this

thanks so much, @ryankshaw!

**Review by jlengstorf — APPROVED** [review]

**Comment by jlengstorf** [comment]
@all-contributors add @ryankshaw for code

**Pull request merged**